### ISSUE TITLE
Can I use framebuffer?

### ISSUE THREAD

**Issue by mionec1** [meefik/linuxdeploy#388]
My phone:XiaoMi MI-ONE C1(Android 4.1.2)
GPU:Adreno 220
CPU:Snapdragon MSM8660
RAM：1GB
I want to install Debian wheezy armhf
Who can tell me?
Thanks a lot!

**Comment by cloud9350** [comment]
Please check this issue
https://github.com/meefik/linuxdeploy/issues/45

or Enter this command from Terminal Emulator
```cat /dev/urandom > /dev/graphics/fb0```
If you view "No space left on device", Your phone is not support Framebuffer.

**Comment by fervi** [comment]
@mionec1 @cloud9350 
Test Cloud is partially incorrect. Please check all exits fb, because it does not have to be a fb0, and the ejection of error does not necessarily mean lack framebuffer.

The procedure looks through the terminal (as root) go to the / dev / graphics and display themselves all the way out fb
ls fb*

Later, using the command
cat /dev/urandom> /dev/fbX

Under the X-type in the number. If you type something will change, ie. There are any dots, artifacts then you probably have access to the framebuffer.

**Comment by cloud9350** [comment]
@fervi This command from https://github.com/meefik/linuxdeploy/wiki/Framebuffer-notes
and I have fb0, fb1, fb2, hdmi but i am not working framebuffer :smile:

**Comment by fervi** [comment]
@cloud9350 

Well, I have since fb0 to fb17 on fb0 I framebuffer, but after entering the command also shows me the error "No space left on device", but displays a different image artifacts. How displays various artifacts, framebuffer will work.

**Comment by cloud9350** [comment]
@fervi OK. I will retry testing.

**Comment by cloud9350** [comment]
@fervi The screen is slightly shaky. What is it?

**Comment by fervi** [comment]
@cloud9350 
Probably. "/dev/urandom" test is not a test which the official. Simply send information to the screen (if you can call it that). If you see something abnormal is a good chance (if not 100%), the framebuffer is working.
Set then Linux Deploy the configuration of the video device, I recommend even set "Freeze Android UI".
Also, remember to double check your "X to include" (best read logs or send) for several days configure X for Arch Linux.

TL;DR - Try it

**Comment by cloud9350** [comment]
@fervi Thanks. Thanks. Very Thanks. :+1: 
You make Bluescreen Burn in my phone. (Enable Freeze Android UI) :100: 
It takes 5 hours to recover. :smile: <-- It's real.

I Android UI Freeze turn off, i can view black screen.
Doesn't work.

**Comment by fervi** [comment]
@cloud9350 
Strange , but as usual you give "pause" does it work ?
If you have black screen it can work, but you don't have XServer or Xterm installed. (hypothesis). You should open /var/log/X* and check where is your problem

**Comment by cloud9350** [comment]
@fervi OK. I will retry.

**Comment by cloud9350** [comment]
@fervi 
Install XServer & XTerm :white_check_mark: 
Android System UI Pause set :white_check_mark: 
I can see SOFT BRICK MY PHONE

"rebooting..."

And Remount, i can see..
![screenshot_2016-02-05-19-55-21](https://cloud.githubusercontent.com/assets/5405019/12844458/a8d81c6c-cc42-11e5-98e6-3a395e7917bb.png)
xorg logging.

Wow.

**Comment by plops** [comment]
We managed to make stuff appear on a screen of a Oneplus One with LineageOS.

Turns out that after every write to the memory mapped area, you have to submit the image for display using ioctl(fbfd, FBIOPUT_VSCREENINFO, &vinfo).

See here for a working code listing with more details:
https://github.com/plops/bacon_fb_test

I compiled it in a chroot debian environment. Make sure to stop surfaceflinger and zygote before running. Otherwise the phone will reboot.
